src/analysis/processing/qgsalgorithmalignsingleraster.cpp

```diff
@@ -0,0 +1,237 @@
+/***************************************************************************
+                         qgsalgorithmalignsingleraster.cpp
+                         ---------------------
+    begin                : July 2023
+    copyright            : (C) 2023 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmalignsingleraster.h"
+#include "qgsprocessingparameteralignrasterlayers.h"
+#include "qgsalignraster.h"
+#include "qgsalignrasterdata.h"
+#include "qgis.h"
+
+///@cond PRIVATE
+
+QgsProcessingAlgorithm::Flags QgsAlignSingleRasterAlgorithm::flags() const
+{
+  return QgsProcessingAlgorithm::flags() | FlagHideFromToolbox;
+}
+
+QString QgsAlignSingleRasterAlgorithm::name() const
+{
+  return QStringLiteral( "alignsingleraster" );
+}
+
+QString QgsAlignSingleRasterAlgorithm::displayName() const
+{
+  return QObject::tr( "Align raster" );
+}
+
+QStringList QgsAlignSingleRasterAlgorithm::tags() const
+{
+  return QObject::tr( "raster,align,resample,rescale" ).split( ',' );
+}
+
+QString QgsAlignSingleRasterAlgorithm::group() const
+{
+  return QObject::tr( "Raster tools" );
+}
+
+QString QgsAlignSingleRasterAlgorithm::groupId() const
+{
+  return QStringLiteral( "rastertools" );
```

Comment:
@alexbruy with this algorithm only available in model designer, shouldn't this group be `modelertools`, instead?

Reply:
As this tool also available in the toolbox, I think it is better to have it the same place in modeler to make it easier for users to find it.